scripts/tableland/tasks.ts

```diff
@@ -69,5 +69,108 @@ task('create-dd-table', 'Creates a Device Definition table related to a Manufact
+            'Manufacturer',
+            instances.localhost.modules.DIMORegistry.address,
+        );
+
+        const tablelandDb = getDatabase(getAccounts()[0]);
+        const tablelandValidator = getValidator(tablelandDb.config.baseUrl);
+
+        console.log(`Minting manufacturers for ${signer.address}...`);
+
+        console.log('Get manufacturers');
+        const manufacturers = (await getDeviceMakes()).data.device_makes;
+        console.log(`Total manufacturers ${manufacturers.length}...`);
+
+        console.log('Get device definitions...');
+        const devices = (await getDeviceDefinitions()).data.device_definitions;
+        console.log(`Total device definitions ${devices.length}...`);
+
+        // manufacturers = manufacturers.slice(0, 5);
+        for await (const element of manufacturers) {
+            const name = `${element.name_slug}`;
+
+            console.log(`Creating Manufacturer ${name} ...`);
+
+            const tx = await manufacturerInstance
+                .connect(signer)
+                .mintManufacturerBatch(signer.address, [name]);
+
+            const receipt = await tx.wait();
+            const manufacturerId = (receipt?.logs[1] as EventLog).args[1].toString();
+
+            console.log(`Manufacturer ${name} minted with ID: ${manufacturerId}`);
+
+            const ddTableInstance: DeviceDefinitionTable = await hre.ethers.getContractAt(
+                'DeviceDefinitionTable',
+                instances.localhost.modules.DIMORegistry.address,
+            );
+
+            console.log(`Creating Device Definition table for manufacturer ${name} with ID  ${manufacturerId}...`);
+
+            const dTx = await ddTableInstance
+                .connect(signer)
+                .createDeviceDefinitionTable(tableOwner, manufacturerId, name);
+
+            await tablelandValidator.pollForReceiptByTransactionHash({
+                chainId: 31337,
+                transactionHash: (await dTx.wait())?.hash as string,
+            });
+
+            const ddTableId = await ddTableInstance.getDeviceDefinitionTableId(manufacturerId);
+            const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId, name);
+
+            console.log(`Device Definition table created\nTable ID: ${ddTableId}\nTable Name: ${ddTableName}`);
+
+            const deviceDefinitionByManufacturers = devices.filter((c)=> c.make.name_slug === element.name_slug && c.type.year > 2006);
+            console.log(`Get Device Definition By Manufacturer [${element.name}] total => ${deviceDefinitionByManufacturers.length}`);
+
+            // deviceDefinitionByManufacturers = deviceDefinitionByManufacturers.slice(0, 5);
+            for await (const dd of deviceDefinitionByManufacturers) {
+                
+                const deviceDefinitionInput : DeviceDefinitionInput = {
+                    id: `${dd.make.name_slug}-${dd.type.model_slug}-${dd.type.year}`,
```

Comment:
Just a reminder to remove the Manufacturer name from the beginning of the table ID. Also, isn't it supposed be `_` instead of `-`?

Reply:
Fixed!!